lib/Sema/TypeCheckType.cpp

```diff
@@ -6146,6 +6146,29 @@ class ExistentialTypeSyntaxChecker : public ASTWalker {
     }
 
     if (auto *proto = dyn_cast<ProtocolDecl>(decl)) {
+      if (proto->existentialRequiresAny()) {
+        //Emit fixit to add any keyword
+        Ctx.Diags.diagnose(T->getNameLoc(), diag::replace_with_any,
+                            proto->getDeclaredInterfaceType(),
+                            proto->getDeclaredExistentialType())
+        .fixItReplace(T->getSourceRange(), "any " + proto->getDeclaredInterfaceType()->getString());
+        //Emit fixit to add some keyword
+        Ctx.Diags.diagnose(T->getNameLoc(), diag::replace_with_some,
+                            proto->getDeclaredInterfaceType(),
+                            ("'some " + proto->getDeclaredInterfaceType()->getString() + "'"))
+        .fixItReplace(T->getSourceRange(), "some " + proto->getDeclaredInterfaceType()->getString());
+        auto diag =
+            Ctx.Diags.diagnose(T->getNameLoc(), diag::existential_requires_any,
+                            proto->getDeclaredInterfaceType(),
+                            proto->getDeclaredExistentialType(),
+                            /*isAlias=*/false);
+        Ctx.Diags.diagnose(T->getNameLoc(), diag::existential_requires_any_or_some,
+                            proto->getDeclaredInterfaceType(),
+                            proto->getDeclaredExistentialType(),
+                            ("'some " + proto->getDeclaredInterfaceType()->getString() + "'"),
+                            /*isAlias=*/false);
+        emitInsertAnyFixit(diag, T);
```

Comment:
This crash from the logs is happening because you are allowed to attach fix-its only to the last emitted diagnostic. You cannot attach a fix-it to a previously emitted diagnostic across new emissions:

```
Assertion `!ActiveDiagnostic && "Already have an active diagnostic"' failed
```

Reply:
so do we want to save the second diagnostic and then emit the any fix it with that one?

Reply:
It must be done in order: emit one diagnostic, attach fix-its, _then_ emit the next one.

Reply:
I see 4 Diagnostics

1. replace_with_any
2. replace_with_some
3. existential_requires_any
4. existential_requires_any_or_some

So each of these require its own `emitInsertAnyFixit`?

Reply:
I checked out `main` and pulled the latest code as well as ran `utils/update-checkout`. I merged main into this feature branch to update everything. However, when I re run the build on the `main` branch, I get no errors. I seem to get errors when running updates on this feature branch and can't understand why. Usually running an entire clean build works, but it isn't anymore.

Reply:
It is a matter of *when* to emit fix-its, not whether to do so. We are not allowed to emit any diagnostics in between emitting `existential_requires_any` (line 6161) and emitting fix-its for `existential_requires_any` (line 6170). I am being specific to make things simpler, but this is a general emission rule.

Reply:
In case that did not help much, let me put it another way. If you need to add a fix-it to an emitted diagnostic, make sure you do so before emitting another diagnostic. If I were to explain this using pseudocode, do it like this:
```
emit(a); addFixits(to: a); emit(b)
```
 Not like this:
 
```
emit(a); emit(b); addFixits(to: a)
```

Reply:
1. Emit (add any keyword) and addFixits(to: add any keyword)

```
Ctx.Diags.diagnose(T->getNameLoc(), diag::replace_with_any,
                    proto->getDeclaredInterfaceType(),
                    proto->getDeclaredExistentialType())
.fixItReplace(T->getSourceRange(), "any " + proto->getDeclaredInterfaceType()->getString());
```

2. Emit (add some keyword) and addFixits(to: add some keyword)

```
Ctx.Diags.diagnose(T->getNameLoc(), diag::replace_with_some,
                    proto->getDeclaredInterfaceType(),
                    ("'some " + proto->getDeclaredInterfaceType()->getString() + "'"))
.fixItReplace(T->getSourceRange(), "some " + proto->getDeclaredInterfaceType()->getString());
```

I am not getting any errors when building with just the code above and it makes sense. 

--------------

Why am I getting errors when I want to emit the following below? Can't seem to understand why because when I opened this PR a while ago, I was getting no errors when building and testing.

3. Emit (Use of protocol 'P' as a type must be written 'any P') with no fix it

```
auto diag =
      Ctx.Diags.diagnose(T->getNameLoc(), diag::existential_requires_any,
                         proto->getDeclaredInterfaceType(),
                         proto->getDeclaredExistentialType(),
                         /*isAlias=*/false);
emitInsertAnyFixit(diag, T);
```

Reply:
> Why am I getting errors when I want to emit the following below?

That highly depends on the errors. Please share them (one at a time, to keep things simple).

> ...with no fix it

`emitInsertAnyFixit` *does* emit fix-its for a given diagnostic. It’s in the name too.

Reply:
> does emit fix-its for a given diagnostic. It’s in the name too.

Sorry I missed that when I copy pasted it. 

---------
I am getting the errors below in multiple files, when I add that `diagnostic: existential_requires_any`

```
error: use of protocol 'IteratorProtocol' as a type must be written 'any IteratorProtocol'
173 |     ///
174 |     internal mutating func resume(
175 |       consuming source: inout some IteratorProtocol<Unicode.Scalar>
    |                                    `- error: use of protocol 'IteratorProtocol' as a type must be written 'any IteratorProtocol'
176 |     ) -> Unicode.Scalar? {
177 |       resume(consuming: { source.next() })
```


```
error: use of protocol 'Sequence' as a type must be written 'any Sequence'
 516 |   @available(SwiftStdlib 6.0, *)
 517 |   public init?<Encoding: Unicode.Encoding>(
 518 |     validating codeUnits: some Sequence<Encoding.CodeUnit>,
     |                                `- error: use of protocol 'Sequence' as a type must be written 'any Sequence'
 519 |     as encoding: Encoding.Type
 520 |   ) {
     :
 578 |   @available(SwiftStdlib 6.0, *)
 579 |   public init?<Encoding>(
 580 |     validating codeUnits: some Sequence<Int8>,
     |                                |- note: replace 'Sequence' with 'any Sequence'
     |                                `- note: replace 'Sequence' with 'some Sequence'
 581 |     as encoding: Encoding.Type
 582 |   ) where Encoding: Unicode.Encoding, Encoding.CodeUnit == UInt8 {
```

Reply:
That seemed to work for me.

1. Switched to `main`, ran the entire build script
2. Switched to `feature` branch and ran `ninja -C ../build/Ninja-RelWithDebInfoAssert/swift-${platform}-$(uname -m) bin/swift-frontend` after making updates

No issues on my end 👍🏽

Reply:
I may be getting this mixed up but it does not seem we need the diagnostic, `existential_requires_any` because we want to use the updated one `existential_requires_any_or_some`. 

We simply recommend 2 fix-its, to replace with `some` or replace with `any`, with only ONE diagnostic suggesting `some` or `any`.

Reply:
Nice job figuring it out. 

> We simply recommend 2 fix-its, to replace with `some` or replace with `any`, with only ONE diagnostic suggesting `some` or `any`.

That makes a lot more sense to me!